core/src/main/scala/org/bitcoins/core/protocol/tlv/TLV.scala

```diff
@@ -278,39 +284,134 @@ object EnumEventDescriptorV0TLV extends TLVFactory[EnumEventDescriptorV0TLV] {
     require(count.toInt == result.size,
             "Did not parse the expected number of outcomes")
 
-    EnumEventDescriptorV0TLV(result)
+    EnumEventDescriptorV0TLV(nonce, result)
   }
 }
 
-case class RangeEventDescriptorV0TLV(start: Int32, stop: Int32, step: UInt16)
+case class RangeEventDescriptorV0TLV(
+    nonce: SchnorrNonce,
+    start: Int32,
+    count: UInt32,
+    step: UInt16)
     extends EventDescriptorTLV {
-  override def tpe: BigSizeUInt = RangeEventDescriptorV0TLV.tpe
+
+  override val tpe: BigSizeUInt = RangeEventDescriptorV0TLV.tpe
 
   override val value: ByteVector = {
-    start.bytes ++ stop.bytes ++ step.bytes
+    nonce.bytes ++ start.bytes ++ count.bytes ++ step.bytes
+  }
+
+  lazy val outcomes: Vector[Int32] = {
+    val startL = start.toLong
+    val stepL = step.toLong
+
+    val range =
+      0L.until(count.toLong).map(num => startL + (num * stepL))
+
+    range.map(Int32(_)).toVector
   }
 }
 
 object RangeEventDescriptorV0TLV extends TLVFactory[RangeEventDescriptorV0TLV] {
 
   override val tpe: BigSizeUInt = BigSizeUInt(55304)
 
+  def apply(
+      nonce: SchnorrNonce,
+      start: Int,
+      count: Int,
+      step: Int): RangeEventDescriptorV0TLV = {
+    RangeEventDescriptorV0TLV(nonce, Int32(start), UInt32(count), UInt16(step))
+  }
+
   override def fromTLVValue(value: ByteVector): RangeEventDescriptorV0TLV = {
     val iter = ValueIterator(value)
 
+    val nonce = SchnorrNonce(iter.take(32))
     val start = Int32(iter.takeBits(32))
-    val stop = Int32(iter.takeBits(32))
+    val count = UInt32(iter.takeBits(32))
     val step = UInt16(iter.takeBits(16))
 
-    RangeEventDescriptorV0TLV(start, stop, step)
+    RangeEventDescriptorV0TLV(nonce, start, count, step)
+  }
+}
+
+case class LargeRangeEventDescriptorV0TLV(
+    base: UInt16,
+    isSigned: Boolean,
```

Comment:
This is built on top of that PR

Reply:
Ok, this would be useful information to include in the original comment. If you add documentation your code reviewers wont ask as many stupid questions :-)

Reply:
Woops, was in the original PRs description. Fixed